### PR TITLE
Birthday field control in user creation endpoint

### LINKED ISSUE #30: Handling Birthday Data
Currently, `Save()` in DAO does nothing with `user.Birthday` since we may not need our users to give their birthday info during the registration process, but `Update()` function should somehow consider about it because user may add their bitday info later.

(Indeed, `Save()` may need also do something with the birthday data from the back-end point of view. If birthday info is not necessary, then front-end can just transfer an empty value.)

However, while `Update()` handling the birthday data, there is a problem:
Currently, `user.Birthday` for illegal or empty value is parsed to `"0001-01-01"` instead of `nil` since a string-type-defined variable `user.Birthday` cannot be `nil` but this might not be the best solution. Therefore, we may need to decide how to handle with the illegal or empty `user.Birthday` value — just store `"0001-01-01"` in the database or store `null` by somehow changing the `user.Birthday` value to `nil`.

### PR DESCRIPTION
**Pull request by JiahuiChen99**
closes #30

### CODE REVIEW

File: domain/users/user_dao.go [hunk 1]
```diff
@@ -40,7 +41,10 @@ func (user *User) Save() *errors.RestErr {
 	// Dates
 	user.DateCreated = date_utils.GetNowString()
 	var birthday time.Time
-	birthday, err = time.Parse(date_utils.DateFormat, user.Birthday)
+	if user.Birthday != "" {
+		birthday, err = time.Parse(date_utils.DateFormat, user.Birthday)
+	}
+
 	if err == nil {
```
Comment (xyzhg) [inline domain/users/user_dao.go]: It might need be inside of block `if user.Birthday != "" {}` (line 44).

**Review by xyzhg — CHANGES_REQUESTED** [review]
`if err == nil {}` may neet to be inside of block `if user.Birthday != "" {}`, otherwise `user.Birthday` will be updated (parsed to `0001-01-01`) anyway.

File: domain/users/user_dao.go (hunk 1 above)
Comment (JiahuiChen99) [reply domain/users/user_dao.go]: Yup, you're right!

**Review by xyzhg — APPROVED** [review]
Good Job~!
Yay!

**Pull request merged**